astroid/brain/brain_builtin_inference.py

```diff
@@ -892,6 +892,26 @@ def _build_dict_with_elements(elements):
     return _build_dict_with_elements([])
 
 
+def _looks_like_copy_method(node):
+    func = node.func
+    return isinstance(func, nodes.Attribute) and func.attrname == "copy"
+
+
+def _has_copy_method(node):
+    return any(
+        isinstance(node, klass)
+        for klass in [nodes.Dict, nodes.List, nodes.Set, objects.FrozenSet]
+    )
+
+
+def _infer_copy_method(node, context=None):
+    inferred = list(node.func.expr.infer(context=context))
```

Comment:
With `list` you evaluate the whole iterator even if the condition would later exit early. The work is done. [itertools.tee()](https://docs.python.org/3/library/itertools.html#itertools.tee) should work here.